state/cloud.go

```diff
@@ -215,6 +216,15 @@ func regionSettingsGlobalKey(cloud, region string) string {
 // If the cloud is in use, ie has models deployed to it, the operation fails.
 func (st *State) RemoveCloud(name string) error {
 	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if _, err := st.Cloud(name); err != nil {
+			// Fail with not found error on first attempt if cloud doesn't exist.
```

Comment:
Any chance you could add a test for that specifically?

Reply:
i did :-)

Reply:
You have added a test that checks that if a cloud is not found, we'd throw this error. 
I am not seeing a test that check for no-op instead for when another (simultaneous) call deleted cloud.

FWIW, I am not convinced that this is really needed. I think this is a bit of an over-engineering and we'd do just as nicely with NotFound always.